src/v/cluster/health_monitor_frontend.cc

```diff
@@ -84,4 +85,20 @@ health_monitor_frontend::get_cluster_health_overview(
         return be.get_cluster_health_overview(deadline);
     });
 }
+
+ss::future<std::error_code>
+health_monitor_frontend::maybe_refresh_cluster_health() {
+    return dispatch_to_backend([](health_monitor_backend& be) mutable {
+        return be.maybe_refresh_cluster_health(
+          force_refresh::no, model::time_from_now(default_timeout));
+    });
+}
+
+ss::future<> health_monitor_frontend::set_disk_health_cb(disk_health_cb_t cb) {
+    return dispatch_to_backend(
```

Comment:
This part is pretty dangerous as it allows to register cross shard callbacks which may lead to all sorts of errors.

Reply:
I vassert() that we are on the same shard in the caller, right?

Reply:
What about doing something similar, but hidden inside health_monitor_frontend? The goal being to avoid the wasted work of building and destroying a cluster health report just to observe the max `disk_space_alert` value.

Reply:
Next update will eliminate the callback. We can redo the optimization inside health monitor frontend if we want in the future.

Reply:
Could I request we keep conversations unresolved until the update has been pushed and is acked to be addressing the feedback?

Reply:
Of course--but we need reviewers to come back and resolve conversations then. Many of us resolve these when we make a commit that addresses the comment--just using it as a to-do list. We should be consistent.

Reply:
Yeah, good point. Hmm, maybe something we should discuss a bit as a team and see what the team finds most helpful/consistent? I've only ever seen the other way, which is that a PR update addresses the comment and let's the original commenter resolve it -- but I could see a bit of judgment involved here with "nit" or "weak preference/maybe suggestions". +1 on being consistent, though.